cmd/policy/cpa_utils.go

```diff
@@ -0,0 +1,103 @@
+package policy
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	yaml "gopkg.in/yaml.v3"
+
+	"github.com/CircleCI-Public/circle-policy-agent/cpa"
+)
+
+//getPolicyDecisionLocally takes path of policy path/directory and input (eg build config) as string, and performs policy evaluation locally
+func getPolicyDecisionLocally(policyPath, input string) (*cpa.Decision, error) {
+	var config interface{}
+	if err := yaml.Unmarshal([]byte(input), &config); err != nil {
+		return nil, fmt.Errorf("invalid config: %w", err)
+	}
+
+	documentBundle, err := getDocumentBundleFromPath(policyPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get document bundle for path: %w", err)
+	}
+
+	parsedPolicy, err := cpa.ParseBundle(documentBundle)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse policy bundle: %w", err)
+	}
+
+	ctx := context.Background()
+	decision, err := parsedPolicy.Decide(ctx, config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to make decision: %w", err)
+	}
+
+	return decision, nil
+}
+
+//getDocumentBundleFromPath takes policy path as an input which could be path to a file or directory, and returns a policy bundle
+//if policyPath is a file, its content will be the only data in the decision bundle
+//if policyPath is a directory, every file in the directory (non-recursive) will be considered a policy in the output policy bundle
+func getDocumentBundleFromPath(policyPath string) (map[string]string, error) {
+	documentBundle := map[string]string{}
+	pathInfo, err := os.Stat(policyPath)
```

Comment:
The stat info approach is not necessary. You may attempt to try and read the directory immediately, if it is not a directory, or does not exist, ReadDir will report that error.